### PR TITLE
for_lane support all configuration files

### PR DESCRIPTION
**Pull request by maschall**
- [x] Run `rspec` for all tools you modified
- [x] Run `rubocop -a` to ensure the code style is valid
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)
#6531

I'm not sure a good reason for the silent ignoring duplicate values in config files, but if you review AppFile, the example given is no longer valid, you can overwrite the same value with duplicate lines.

Also I started with the tests for Gymfile but realized quickly that the configuration effects Dangerfile, Matchfile, Scanfile, etc...

I can help update docs if this looks good

### CODE REVIEW

**Comment by KrauseFx** [comment]
Oh wow, thanks for the pull request, I somehow completely missed that notification. Definitely something we should look into.

File: fastlane_core/spec/configuration_file_spec.rb [hunk 1]
```diff
@@ -64,12 +64,6 @@
         expect(config[:another_boolean]).to be(true)
       end
 
-      it "ignores the same item being set after the first time" do
```
Comment (KrauseFx) [inline fastlane_core/spec/configuration_file_spec.rb]: By changing this, would this break setups where you have the same line in one config file twice, and you are used to them being ignored? Is there a way we can leave that test and make it pass, while still adding this new feature?

File: fastlane_core/spec/fixtures/ConfigFileForLane [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  app_identifier "com.forlane.enterprise"
+end
+
+for_platform "ios" do
+  app_identifier "com.forplatform.ios"
+  
+  for_lane "release" do
+    app_identifier "com.forplatformios.forlanerelease"
+  end
+end
```
Comment (KrauseFx) [inline fastlane_core/spec/fixtures/ConfigFileForLane]: Can you add a new line here

**Review by KrauseFx — CHANGES_REQUESTED** [review]
Great tests, great change, I love it 👍 

Please add some documentation somewhere, probably in Advanced.md for now, and we're good to go. 

Sorry it took so long, and thanks again for working on this :rocket:

File: fastlane_core/spec/configuration_file_spec.rb (hunk 1 above)
Comment (maschall) [inline fastlane_core/spec/configuration_file_spec.rb]: I was wondering about this, I'll take another stab at it, but I believe (and may be wrong) is that Appfile is not behaving in this same manner, and that was why I looked at removing this.

I'll see what I can do to get this back in there.

**Comment by asfalcone** [comment]
@maschall It looks like there are a few changes still needed for this PR.

File: fastlane_core/spec/fixtures/ConfigFileForLane (hunk 2 above)
Comment (mfurtak) [inline fastlane_core/spec/fixtures/ConfigFileForLane]: Done

**Comment by mfurtak** [comment]
Since it seems like we were still pretty keen on getting this merged, I did some more work on it in #7859 to reconcile what I perceived to be the blocking problems here.

**Comment by mfurtak** [comment]
Carried forward and merged as part of #7859 👍

**Pull request closed without merging**